Sources/Combine/Paging/Protocol/Page.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  Page.swift
+//  Catalog
+//
+//  Created by Antonijo Bezmalinovic on 10.01.2023..
+//  Copyright (c) 2021 Infinum. All rights reserved.
+//
+
+import Foundation
+
+/// Defines template for page
+protocol Page where Item: Pageable {
+    
+    /// Constraint results to types that conform to Pageable
+    associatedtype Item
+    
+    var count: Int { get }
+    var next: URL? { get }
+    var previous: URL? { get }
+    var pages: Int { get }
+    var page: Int { get }
+    var results: [Item] { get }
+}
```

Comment:
Is there a reason why you went for an implementation where you would constrain it with `where` instead of just making it generic? Then you wouldn't need the `associatedType` if I'm not mistaken 😄 
```suggestion
/// Defines template for page
protocol Page<Item: Pageable> { 
    var count: Int { get }
    var next: URL? { get }
    var previous: URL? { get }
    var pages: Int { get }
    var page: Int { get }
    var results: [Item] { get }
}
```

Reply:
Da i ja sam se oko toga lomio, zapravo oba nacina su legit i raditi ce, pa sam isao malo istrazivati kada bi sto trebalo koristiti, ugl. zakljucak je ako zelis constrainati protocol na nacin da mora koristiti bilo koji tip koji conforma nekom drugom protokolu da je onda ispravan nacin koristiti assossiated types, dok bi generics vise isao u prilog kada bi imao protokol koji zelis da conforma nekom manjem setu zadanih tipova. Na kraju sam pitao ChatGPT :), i evo odgovora: 

"Associated types are useful when you want to define a protocol that can work with any type, but you want to ensure that the types that conform to the protocol have certain properties or methods. For example, in the above example, you want to ensure that the type that conforms to the Page protocol is also conforms to Pageable protocol.

Generic types are useful when you want to create a more specific protocol that works with one or a few specific types. For example, you have a struct that works with a specific model, and you want to add protocol conformance for it."

Sto ti kazes, koji su cons a koji pros za svaki od pristupa?